bliss-app/src/components/bliss/bliss-index.jsx

```diff
@@ -3,95 +3,125 @@ import BlissList from "./bliss-list";
 import BlissCreateForm from "./bliss-create-form";
 import { useDispatch, useSelector } from "react-redux";
 import {
-	allBliss,
-	blissError,
-	blissStatus,
-	cleanBliss,
-	fetchBliss,
-	prevPage,
-	nextPage,
+  allBliss,
+  blissError,
+  blissStatus,
+  cleanBliss,
+  fetchBliss,
+  prevPage,
+  nextPage,
 } from "../../store/blissSlice";
 import { useSearchParams } from "react-router-dom";
 
-function BlissPage() {
-	const dispatch = useDispatch();
-	const [qparams, setqparams] = useSearchParams();
-	const [limit, setLimit] = useState(15);
-	const [page, setPage] = useState(parseInt(qparams.get("page")));
-	const bliss = useSelector(allBliss);
-	const status = useSelector(blissStatus);
-	const error = useSelector(blissError);
-	const isMounted = useRef(false);
-	let content;
-	const [addBliss, setAddBliss] = useState(false);
+/**
+ * Parse a valid page number from a URLSearchParams object.
+ *
+ * Invalid states, empty states, and NaN are treated as the first page of results.
+ */
+const parsePageFromQueryParams = (query) => {
+  let initialPage = 1;
+  const pageFromQuery = query.get("page");
+  if (pageFromQuery) {
+    initialPage = parseInt(pageFromQuery);
+    if (isNaN(initialPage)) {
+      initialPage = 1;
+    }
+  }
+
+  return initialPage;
+};
+
+/**
+ * Puts page params logic into a hook.
+ *
+ * This will re-run when the query parameters change.
+ *
+ * It returns functions that will change the query parameters to update the page correctly, as well.
+ */
+const usePagination = () => {
+  const limit = 15;
+  const [qparams, setqparams] = useSearchParams();
+
+  const page = parsePageFromQueryParams(qparams);
+
+  const onClickNext = () => {
+    setqparams({ ...qparams, page: page + 1 });
```

Comment:
In this model, changes happen by setting the query params or reading the query params.  The existing hook will take care of re-running this function whenever the user changes the URL (or whenever your code changes the URL.)